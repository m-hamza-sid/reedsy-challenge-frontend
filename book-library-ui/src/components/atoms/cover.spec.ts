import { mount } from "@vue/test-utils";
import { describe, expect, test } from "vitest";
import Cover from "./cover.vue";

describe("cover", () => {
  test("should render in small size by default", () => {
    const iSrc = "https://some-cover.png";
    const wrapper = mount(Cover, { attrs: { src: iSrc } });
    expect(wrapper.attributes()).toContain({ src: iSrc });
    expect(wrapper.classes()).toContain("w-32");
  });
  test("should render in large size if size prop is not small", () => {
    const iSrc = "https://some-cover.png";
    const wrapper = mount(Cover, { attrs: { src: iSrc }, props: { size: "large" } });
    expect(wrapper.attributes()).toContain({ src: iSrc });
    expect(wrapper.classes()).toContain("w-72");
  });
  test("should match default snapshot", () => {
    const iSrc = "https://some-cover.png";
    const wrapper = mount(Cover, { attrs: { src: iSrc } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
