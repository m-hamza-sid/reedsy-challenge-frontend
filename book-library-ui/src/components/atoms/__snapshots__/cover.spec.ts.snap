// Vitest Snapshot v1

exports[`cover > should match default snapshot 1`] = `"<img class=\\"rounded-lg w-32\\" src=\\"https://some-cover.png\\" alt=\\"\\">"`;
